### ISSUE TITLE
We need to hide more kernel functions

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#1734]
When tracing kernel functions, I see too many unrelated kernel functions in the trace. It might be because of security policies or some other reasons, but it makes the tracing output too much complicated unnecessarily.

For example, i see too many kernel functions in the following list.
- `down_read_trylock`, `up_read`,  `__cond_resched`, `irq_enter_rcu`, `down_read`

I also see too many `find_vma` and `handle_mm_fault` from recent kernels compared to the old one.

To reduce the confusion, I would like to have a way to control the verbosity level of kernel functions.

Currently, we have only the option for kernel scope with `--kernel-full`, but would like to have another way to hide most of those functions, but show only when users want.

**Comment by honggyukim** [comment]
For example, a simple hello world program shows the following trace output.
```
$ sudo uftrace -k hello
Hello World!
# DURATION     TID     FUNCTION
   0.295 us [  8854] | __monstartup();
   0.096 us [  8854] | __cxa_atexit();
            [  8854] | main() {
            [  8854] |   puts() {
   0.164 us [  8854] |     down_read_trylock();
   0.148 us [  8854] |     __cond_resched();
   0.316 us [  8854] |     find_vma();
   2.124 us [  8854] |     handle_mm_fault();
   0.145 us [  8854] |     up_read();
   4.640 us [  8854] |     __x64_sys_newfstatat();
  13.228 us [  8854] |     __x64_sys_write();
  25.276 us [  8854] |   } /* puts */
   0.222 us [  8854] |   fflush();
  25.929 us [  8854] | } /* main */
```

**Comment by honggyukim** [comment]
Maybe we can provide an option something like `--kernel-verbosity` but it might be not easy to categorize which functions are verbose functions and which are not.

**Comment by namhyung** [comment]
Yeah, actually [it works like below](https://elixir.bootlin.com/linux/v6.1/source/arch/x86/mm/fault.c#L1507) (for user page faults, v6.1 kernel):

```
exc_page_fault() {          /* noinstr */
  handle_page_fault() {     /* always inline */
    do_user_addr_fault() {  /* inline */
      down_read_trylock();
      __cond_resched();
      find_vma();
      handle_mm_fault();
      up_read();
    }
  }
}
```

If one of the first 3 functions was trace-able, it's fine (like it did with `__page_fault()` which is declared as `noinline` until v5.6) since we apply depth of 1 for kernel functions with `-k`, but now all these functions are either non-instrumented or inlined.  So we see the internal functions separately.
